lms/djangoapps/experiments/flags.py

```diff
@@ -266,21 +266,6 @@ def get_bucket(self, course_key=None, track=True):
             # Mark that we've recorded this bucketing, so that we don't do it again this session
             request.session[session_key] = True
```

Comment:
This gets taken out because it's now a WaffleFlag, and not an ExperimentWaffleFlag, correct?

Reply:
I don't think I understand what this code is doing, It is bucketing users, and then sending a message if the user is eligible for inclusion in the streak discounting, but only when they have gotten a three day streak? This is just used to build the tranche of users who could possibly see the discount? 
is "edx.bi.course.streak_discount_enabled" a reasonable name for the event?

Reply:
I put this code back in, but it never seems to be reached. I put a breakpoint in the bucketing code, and it is never hit. If I have changed the flag from a WaffleExperiment to a WaffleFlag, does this event need to be generated somewhere else?

Reply:
ah yeah i forgot this code was in get bucket, you want to move the code elsewhere, but you do want to keep firing an event
that event name seems fine to me

Reply:
ok, moved it to the discount calculation in get_celebrations_dict